apps/remote_control/lib/lexical/remote_control/port.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Lexical.RemoteControl.Port do
+  @moduledoc """
+  Utilities for launching ports in the context of a project
+  """
+
+  alias Lexical.Project
+  alias Lexical.RemoteControl
+
+  @type open_opt ::
+          {:env, list()}
+          | {:cd, String.t() | string()}
+          | {:env, [{:os.env_var_name(), :os.env_var_value()}]}
+          | {:args, list()}
+
+  @type open_opts :: [open_opt]
+
+  @doc """
+  Launches elixir in a port.
+
+  This function takes the project's context into account and looks for the executable via calling
+  `RemoteControl.elixir_executable(project)`. Environment variables are also retrieved with that call.
+  """
+  @spec open_elixir(Project.t(), open_opts()) :: port()
+  def open_elixir(%Project{} = project, opts) do
+    {:ok, elixir_executable, environment_variables} = RemoteControl.elixir_executable(project)
+
+    opts =
+      opts
+      |> Keyword.put_new_lazy(:cd, fn -> Project.root_path(project) end)
+      |> Keyword.put_new(:env, environment_variables)
+
+    open(project, elixir_executable, opts)
+  end
+
+  @doc """
+  Launches an executable in the project context via a port.
+  """
+  def open(%Project{} = project, executable, opts) do
+    {launcher, opts} = Keyword.pop_lazy(opts, :path, &path/0)
+
+    opts =
+      opts
+      |> Keyword.put_new_lazy(:cd, fn -> Project.root_path(project) end)
+      |> Keyword.update(:args, [executable], fn old_args -> [executable | old_args] end)
```

Comment:
Ensure string args (see comment on usage in [`connection.ex`](apps/remote_control/lib/lexical/remote_control/search/store/mnesia/connection.ex))

```suggestion
      |> Keyword.update(:args, [executable], fn old_args ->
        Enum.map([executable | old_args], &to_string/1)
      end)
```

Reply:
I'm not sure how I feel about this. 

Things can get turned into strings in surprising ways. I favor ensuring the caller does what they want to do with the arguments and pass them into this function.